bvaughn/react-devtools-experimental
# Automated tests [WIP]

Relates to #115

Test goals:
* Write test code using a real React renderer (e.g. ReactDOM) injected into the real DevTools hook.
* Use the real renderer interface to generate tree op codes and send them through a real `Bridge` to a real `Store`.
* Use Jest snapshot mechanism with a pretty-printed version of the `Store` tree.

So a test like this:
```js
const Grandparent = () => (
  <React.Fragment>
    <Parent />
    <Parent />
  </React.Fragment>
);
const Parent = () => [<Child key="one" />, <Child key="two" />];
const Child = () => <div>Hi!</div>;

ReactDOM.render(<Grandparent />, document.createElement("div"));

expect(store).toMatchSnapshot();
```

Will generate a snapshot like this:
```
[root]
  ▾ <Grandparent>
    ▾ <Parent>
        <Child key="one">
        <Child key="two">
    ▾ <Parent>
        <Child key="one">
        <Child key="two">
```
